### PR TITLE
[SH4RE-29] Made Gap Disappear for View Listing and Small Changes

### PR DESCRIPTION
**Pull request by hannahbrooks**

### CODE REVIEW

File: SH4RE/SH4RE/ViewListing/ViewListingView.swift [hunk 1]
```diff
@@ -34,62 +34,66 @@ struct ViewListingView: View {
...
-                
-                Text(self.description)
-                    .font(.body)
-                    .padding([.horizontal])
-                    .padding([.top], 10)
-                
-                Button(action: {
-                    print("Check Availability button clicked")
-                }) {
+                    Color.white.frame(width: screenSize.width, height: 275).opacity(0)
```
Comment (hannahbrooks) [inline SH4RE/SH4RE/ViewListing/ViewListingView.swift]: i added this. a lil hacky but whatevs

File: SH4RE/SH4RE/ViewListing/ViewListingView.swift [hunk 2]
```diff
@@ -34,62 +34,66 @@ struct ViewListingView: View {
...
-                
-                GeometryReader { geometry in
-                    ImageCarouselView(numberOfImages: self.numberOfImages) {
-                        ForEach([self.image, UIImage(named: "ProfilePhotoPlaceholder")!, UIImage(named: "ProfilePhotoPlaceholder")!], id:\.self) { image in
-                            Image(uiImage: image)
-                                .resizable()
-                                .scaledToFit()
-                                .frame(width: geometry.size.width, height: 250)
-                                .aspectRatio(contentMode: .fill)
+            ScrollView {
```
Comment (hannahbrooks) [inline SH4RE/SH4RE/ViewListing/ViewListingView.swift:34]: added scrollview here

Comment (americobarros) [reply SH4RE/SH4RE/ViewListing/ViewListingView.swift:34]: nice, looks good

File: SH4RE/SH4RE/ViewListing/ViewListingView.swift (hunk 1 above)
Comment (americobarros) [reply SH4RE/SH4RE/ViewListing/ViewListingView.swift]: Theres another way to do it that isn't hacky

File: SH4RE/SH4RE/ViewListing/ViewListingView.swift [hunk 3]
```diff
@@ -34,62 +34,66 @@ struct ViewListingView: View {
...
+                        ImageCarouselView(numberOfImages: self.numberOfImages) {
+                            ForEach([self.image, UIImage(named: "ProfilePhotoPlaceholder")!, UIImage(named: "ProfilePhotoPlaceholder")!], id:\.self) { image in
+                                Image(uiImage: image)
+                                    .resizable()
+                                    .scaledToFit()
+                                    .frame(width: geometry.size.width, height: 250)
+                                    .aspectRatio(contentMode: .fill)
+                            }
                         }
                     }
```
Comment (americobarros) [inline SH4RE/SH4RE/ViewListing/ViewListingView.swift:47]: add `.frame(height:295)` at line 51. [See Here](https://www.hackingwithswift.com/quick-start/swiftui/how-to-provide-relative-sizes-using-geometryreader)

File: SH4RE/SH4RE/ViewListing/ViewListingView.swift [hunk 4]
```diff
@@ -34,62 +34,66 @@ struct ViewListingView: View {
...
-                    .font(.body)
-                    .padding([.horizontal])
-                    .padding([.top], 10)
-                
-                Button(action: {
-                    print("Check Availability button clicked")
-                }) {
+                    Color.white.frame(width: screenSize.width, height: 275).opacity(0)
+                    
+                    Text(self.title)
```
Comment (americobarros) [inline SH4RE/SH4RE/ViewListing/ViewListingView.swift]: is this wasifs code? why all the self?

**Review by americobarros — CHANGES_REQUESTED** [review]
Looks good, i would say change to the frame tho, less hacky

File: SH4RE/SH4RE/ViewListing/ViewListingView.swift (hunk 4 above)
Comment (americobarros) [reply SH4RE/SH4RE/ViewListing/ViewListingView.swift]: [Heres what they say about using self](https://dmitripavlutin.com/how-to-use-correctly-self-keyword-in-swift/#:~:text=The%20answer%20is%20that%20Swift%20permits%20and%20even%20encourages%20you%20to%20omit%20self%20keyword%20when%20it%20can%20be%20done.%20The%20challenge%20is%20to%20determine%20the%20scenarios%20when%20self%20is%20obligatory%20and%20when%20optional)

Comment (hannahbrooks) [reply SH4RE/SH4RE/ViewListing/ViewListingView.swift]: i changed!

**Review by americobarros — APPROVED** [review]

**Pull request merged**